### PR TITLE
CM-22829 - Migrate to Poetry; fix package structure; use dynamic versioning from Git Tags

### PR DESCRIPTION
**Pull request by MarshalX**
- [x] add dynamic version from one source (from VCS tags)

### CODE REVIEW

**Comment by MarshalX** [comment]
Some useful info to understand all things better

**Including config files to the package**

> Unlike setuptools poetry bundles automatically all files within your package folder into your package

It means our config.yml files are already included by default. I test it locally by run "poetry build" and unarchive the dist files. it works

**We should not use data_files anymore**

https://github.com/cycodehq-public/cycode-cli/blob/a0ca07f91cef2ea333f21c0d58dd58f32f39c0b4/setup.py#L18
why:
1. deprecated. ref: https://setuptools.pypa.io/en/latest/references/keywords.html
image

3. relates only to setuptools
4. not recommended at all ([include_package_data](https://setuptools.pypa.io/en/latest/userguide/datafiles.html) should be used instead)

**We don't use setuptools anymore**

Poetry core fully covers our requirements. It automatically understand type of README, fetch the content from the file and so on

**We don't distribute 2 packages inside one PyPI project**

Previously we package contained 2 packages: cli, cyclient. This is could lead to problems. Let's take a look how our package installed in the system (pip install cycode && pip uninstall cycode):

image

as we can see it creates 2 folders in the global system space for all Python packages. to be more specific to "site-packages" folder. And it's named CLI

image

The name doesn't relate to cycode at all. Any other packages named "cli" will break something due to conflicts in the file system :( 

also, I want to mention, that if we want to provide a separate cyclient package we should distribute it separately on PyPi (in case we want to provide opensource lib to communicate with out backend from a code. I mean without CLI) 

**That's how the tree looks now after restructure**

image

**Comment by MarshalX** [comment]
Some useful info to understand all things better

**Including config files to the package**

> Unlike setuptools poetry bundles automatically all files within your package folder into your package

It means our config.yml files are already included by default. I test it locally by run "poetry build" and unarchive the dist files. it works

**We should not use data_files anymore**

https://github.com/cycodehq-public/cycode-cli/blob/a0ca07f91cef2ea333f21c0d58dd58f32f39c0b4/setup.py#L18
why:
1. deprecated. ref: https://setuptools.pypa.io/en/latest/references/keywords.html
image

3. relates only to setuptools
4. not recommended at all ([include_package_data](https://setuptools.pypa.io/en/latest/userguide/datafiles.html) should be used instead)

**We don't use setuptools anymore**

Poetry core fully covers our requirements. It automatically understand type of README, fetch the content from the file and so on

**We don't distribute 2 packages inside one PyPI project**

Previously we package contained 2 packages: cli, cyclient. This is could lead to problems. Let's take a look how our package installed in the system (pip install cycode && pip uninstall cycode):

image

as we can see it creates 2 folders in the global system space for all Python packages. to be more specific to "site-packages" folder. And it's named CLI

image

The name doesn't relate to cycode at all. Any other packages named "cli" will break something due to conflicts in the file system :( 

also, I want to mention, that if we want to provide a separate cyclient package we should distribute it separately on PyPi (in case we want to provide opensource lib to communicate with out backend from a code. I mean without CLI) 

**That's how the tree looks now after restructure**

image

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+name = "cycode"
+version = "0.2.3" # TODO(MarshalX): Dynamic from VCS
+description = "Perform secrets/iac scans for your sources using Cycode's engine"
+keywords=["secret-scan", "cycode", "devops", "token", "secret", "security", "cycode", "code"]
+authors = ["Cycode <support@cycode.com>"]
+license = "MIT"
+repository = "https://github.com/cycodehq-public/cycode-cli"
+readme = "README.md"
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
```
Comment (MarshalX) [inline pyproject.toml:11]: What is the actual status of our CLI? @artem-fedorov @MichalBor 

here is the list of available options:
[Development Status :: 1 - Planning](https://pypi.org/search/?c=Development+Status+%3A%3A+1+-+Planning) 
 [Development Status :: 2 - Pre-Alpha](https://pypi.org/search/?c=Development+Status+%3A%3A+2+-+Pre-Alpha) 
[Development Status :: 3 - Alpha](https://pypi.org/search/?c=Development+Status+%3A%3A+3+-+Alpha)   
[Development Status :: 4 - Beta](https://pypi.org/search/?c=Development+Status+%3A%3A+4+-+Beta)   
[Development Status :: 5 - Production/Stable](https://pypi.org/search/?c=Development+Status+%3A%3A+5+-+Production%2FStable)  
[Development Status :: 6 - Mature](https://pypi.org/search/?c=Development+Status+%3A%3A+6+-+Mature)  
[Development Status :: 7 - Inactive](https://pypi.org/search/?c=Development+Status+%3A%3A+7+-+Inactive)

**Comment by MichalBor** [comment]
> Some useful info to understand all things better
> 
> **Including config files to the package**
> 
> > Unlike setuptools poetry bundles automatically all files within your package folder into your package
> 
> It means our config.yml files are already included by default. I test it locally by run "poetry build" and unarchive the dist files. it works
> 
> **We should not use data_files anymore**
> 
> https://github.com/cycodehq-public/cycode-cli/blob/a0ca07f91cef2ea333f21c0d58dd58f32f39c0b4/setup.py#L18
> 
> 
> why:
> 1. deprecated. ref: https://setuptools.pypa.io/en/latest/references/keywords.html
> 
> image
> 3. relates only to setuptools
> 4. not recommended at all ([include_package_data](https://setuptools.pypa.io/en/latest/userguide/datafiles.html) should be used instead)
> 
> **We don't use setuptools anymore**
> 
> Poetry core fully covers our requirements. It automatically understand type of README, fetch the content from the file and so on
> 
> **Now we distribute 1 package inside of one PyPI project**
> 
> Previously our PyPi project contained 2 packages: cli, cyclient. This could lead to problems. Let's take a look at how our package is installed in the system (pip install cycode && pip uninstall cycode):
> 
> image
> as we can see it creates 2 folders in the **global system space** for all Python packages. to be more specific to "site-packages" folder. And it's named "cli"
> 
> image
> The name doesn't relate to cycode at all. Any other packages named "cli" will break something due to conflicts in the file system :(
> 
> also, I want to mention, that if we want to provide a separate cyclient package we should distribute it separately on PyPi (in case we want to provide Open Source lib to communicate with our backend from a code. I mean without CLI)
> 
> **That's how the tree looks now after restructuring**
> 
> image
> But it breaks imports. @MichalBor let me know if some of our projects uses CLI by importing it directly in the code instead of interacting with command line interface
> 
> Rework of versioning is coming soon. More details in the task

👑

File: poetry.lock [hunk 2]
```diff
@@ -0,0 +1,788 @@
...
+category = "dev"
+optional = false
+python-versions = ">=3.7"
+files = [
+    {file = "packaging-23.1-py3-none-any.whl", hash = "sha256:994793af429502c4ea2ebf6bf664629d07c1a9fe974af92966e4b8d2df7edc61"},
+    {file = "packaging-23.1.tar.gz", hash = "sha256:a392980d2b6cffa644431898be54b0045151319d1e7ec34f0cfed48767dd334f"},
+]
+
+[[package]]
+name = "pathspec"
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply poetry.lock:428]: Hi @MaorDavidzon. What should we do? Our CLI uses it directly. The right version of the source is https://github.com/cpburnz/python-pathspec

File: poetry.lock [hunk 3]
```diff
@@ -0,0 +1,788 @@
...
+files = [
+    {file = "binaryornot-0.4.4-py2.py3-none-any.whl", hash = "sha256:b8b71173c917bddcd2c16070412e369c3ed7f0528926f70cac18a6c97fd563e4"},
+    {file = "binaryornot-0.4.4.tar.gz", hash = "sha256:359501dfc9d40632edc9fac890e19542db1a287bbcfa58175b66658392018061"},
+]
+
+[package.dependencies]
+chardet = ">=3.0.2"
+
+[[package]]
+name = "certifi"
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply poetry.lock:67]: "certifi" is the dependency of "requests" lib that we are using in CLI.  Is it a false positive? @MaorDavidzon

File: poetry.lock [hunk 4]
```diff
@@ -0,0 +1,788 @@
...
+category = "main"
+optional = false
+python-versions = ">=3.6"
+files = [
+    {file = "certifi-2023.5.7-py3-none-any.whl", hash = "sha256:c6c2e98f5c7869efca1f8916fed228dd91539f9f1b444c314c06eef02980c716"},
+    {file = "certifi-2023.5.7.tar.gz", hash = "sha256:0f0d56dc5a6ad56fd4ba36484d6cc34451e1c6548c61daad8c320169f91eddc7"},
+]
+
+[[package]]
+name = "chardet"
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply poetry.lock:79]: same as "certifi"

File: poetry.lock (hunk 3 above)
[the remark this answers is not in the document]
Comment (MarshalX) [reply poetry.lock:34]: btw the requests lib under Apache 2.0 and uses MPL under the hood

File: poetry.lock [hunk 5]
```diff
@@ -0,0 +1,739 @@
...
+files = [
+    {file = "binaryornot-0.4.4-py2.py3-none-any.whl", hash = "sha256:b8b71173c917bddcd2c16070412e369c3ed7f0528926f70cac18a6c97fd563e4"},
+    {file = "binaryornot-0.4.4.tar.gz", hash = "sha256:359501dfc9d40632edc9fac890e19542db1a287bbcfa58175b66658392018061"},
+]
+
+[package.dependencies]
+chardet = ">=3.0.2"
+
+[[package]]
+name = "certifi"
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply poetry.lock:34]: #cycode_ignore_non_permissive_license_use

File: poetry.lock [hunk 6]
```diff
@@ -0,0 +1,739 @@
...
+category = "main"
+optional = false
+python-versions = ">=3.6"
+files = [
+    {file = "certifi-2023.5.7-py3-none-any.whl", hash = "sha256:c6c2e98f5c7869efca1f8916fed228dd91539f9f1b444c314c06eef02980c716"},
+    {file = "certifi-2023.5.7.tar.gz", hash = "sha256:0f0d56dc5a6ad56fd4ba36484d6cc34451e1c6548c61daad8c320169f91eddc7"},
+]
+
+[[package]]
+name = "chardet"
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply poetry.lock:46]: #cycode_ignore_non_permissive_license_use

File: poetry.lock [hunk 7]
```diff
@@ -0,0 +1,739 @@
...
+category = "dev"
+optional = false
+python-versions = ">=3.7"
+files = [
+    {file = "packaging-23.1-py3-none-any.whl", hash = "sha256:994793af429502c4ea2ebf6bf664629d07c1a9fe974af92966e4b8d2df7edc61"},
+    {file = "packaging-23.1.tar.gz", hash = "sha256:a392980d2b6cffa644431898be54b0045151319d1e7ec34f0cfed48767dd334f"},
+]
+
+[[package]]
+name = "pathspec"
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply poetry.lock:410]: #cycode_ignore_non_permissive_license_use

**Review by artem-fedorov — APPROVED** [review]

**Pull request merged**